controllers/datadogagent/feature/helmcheck/feature.go

```diff
@@ -0,0 +1,188 @@
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/common"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils/comparison"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
+)
+
+func init() {
+	err := feature.Register(feature.HelmCheckIDType, buildHelmCheckFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildHelmCheckFeature(options *feature.Options) feature.Feature {
+	helmCheckFeat := &helmCheckFeature{
+		rbacSuffix: common.ClusterAgentSuffix,
+	}
+
+	if options != nil {
+		helmCheckFeat.logger = options.Logger
+	}
+	return helmCheckFeat
+}
+
+type helmCheckFeature struct {
+	runInClusterChecksRunner bool
+	collectEvents            bool
+	valuesAsTags             map[string]string
+
+	serviceAccountName string
+	rbacSuffix         string
+
+	owner                       metav1.Object
+	customConfig                *apicommonv1.CustomConfig
+	configMapName               string
+	customConfigAnnotationKey   string
+	customConfigAnnotationValue string
+
+	logger logr.Logger
+}
+
+// ID returns the ID of the Feature
+func (f *helmCheckFeature) ID() feature.IDType {
+	return feature.HelmCheckIDType
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *helmCheckFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	f.owner = dda
+	helmCheck := dda.Spec.Features.HelmCheck
+
+	if helmCheck != nil && apiutils.BoolValue(helmCheck.Enabled) {
+		reqComp.ClusterAgent.IsRequired = apiutils.NewBoolPointer(true)
+		reqComp.Agent.IsRequired = apiutils.NewBoolPointer(true)
+
+		if helmCheck.Conf != nil {
+			f.customConfig = v2alpha1.ConvertCustomConfig(helmCheck.Conf)
+			hash, err := comparison.GenerateMD5ForSpec(f.customConfig)
+			if err != nil {
+				f.logger.Error(err, "couldn't generate hash for helm check custom config")
+			} else {
+				f.logger.V(2).Info("built helm check from custom config", "hash", hash)
+			}
+			f.customConfigAnnotationKey = object.GetChecksumAnnotationKey(feature.HelmCheckIDType)
+			f.customConfigAnnotationValue = hash
+		}
+
+		f.configMapName = apicommonv1.GetConfName(dda, f.customConfig, apicommon.DefaultHelmCheckConf)
+		f.collectEvents = apiutils.BoolValue(helmCheck.CollectEvents)
+		f.valuesAsTags = helmCheck.ValuesAsTags
+		f.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+
+		if v2alpha1.IsClusterChecksEnabled(dda) && v2alpha1.IsCCREnabled(dda) {
```

Comment:
Should we add customConf flag here as well in order to override `reqComp` on line 98?

Reply:
Removed customConf to maintain parity with helm chart